tangram_bundler/__init__.py

```diff
@@ -1,100 +1,235 @@
+def addUniformTextureDependency(fileList, basePath, rootNode, styleName, uniformName):
     referenceTexturePath = ""
     explicitUniformTexturePath = ""
-    key = yaml_file['styles'][styleName]['shaders']['uniforms'][uniformName]
+    key = rootNode['styles'][styleName]['shaders']['uniforms'][uniformName]
 
     if isinstance(key, basestring):
-        if ('textures' in yaml_file) and (key in yaml_file['textures']):
-            referenceTexturePath = basePath + '/' + yaml_file['textures'][ yaml_file['styles'][styleName]['shaders']['uniforms'][uniformName] ]['url']
-        explicitUniformTexturePath = basePath + '/' + yaml_file['styles'][styleName]['shaders']['uniforms'][uniformName]
+        if ('textures' in rootNode) and (key in rootNode['textures']):
+            referenceTexturePath = os.path.relpath(rootNode['textures'][ rootNode['styles'][styleName]['shaders']['uniforms'][uniformName] ]['url'], basePath)
+        explicitUniformTexturePath = os.path.relpath(rootNode['styles'][styleName]['shaders']['uniforms'][uniformName], basePath)
 
     if (os.path.exists(explicitUniformTexturePath)):
-        file_list.append(os.path.normpath(explicitUniformTexturePath))
+        fileList.append(explicitUniformTexturePath)
     elif (os.path.exists(referenceTexturePath)):
-        file_list.append(os.path.normpath(referenceTexturePath))
-
-
-# Append yaml dependences in yaml_file ('import' files) to another yaml file (full_yaml_file)
-def addDependencies(file_list, filename):
-    print "Adding dependencies for file: ", filename
-    file_list.append(os.path.normpath(filename))
-    folder = os.path.dirname(filename)
-    yaml_file = yaml.safe_load(open(filename))
-
-    # TODO:
-    #   - Check if any url actually exist and is local before doing nothing
+        fileList.append(referenceTexturePath)
 
+# Fetch all asset dependencies in the constructed rootNode
+def fetchDependencies(fileList, rootNode, basePath):
     # Search for fonts urls
-    if 'fonts' in yaml_file:
-        for font in yaml_file['fonts']:
-            # initial Tangram font support
-            if 'url' in yaml_file['fonts'][font]:
-                file_list.append(os.path.normpath(folder + '/' + yaml_file['fonts'][font]['url']))
-
-            # newer Tangram font support
-            for weight in yaml_file['fonts'][font]:
-                if 'url' in weight:
-                    file_list.append(os.path.normpath(folder + '/' + weight['url']))
+    if 'fonts' in rootNode:
+        fontsNode = rootNode['fonts']
+        for font in fontsNode:
+            fontNode = fontsNode[font]
+            for prop in fontNode:
+                # initial Tangram font support
+                if 'url' in prop:
+                    fileList.append(os.path.relpath(prop['url'], basePath))
+                # newer Tangram font support
+                if 'weight' in prop and type(prop['weight']) is str:
+                    fileList.append(os.path.relpath(prop['weight'], basePath))
```

Comment:
I'm not clear on what's supposed to be happening here. When `fonts` is a single object, a single `url` will specify a font file to always be used for that family. When `fonts` is an **array of objects**, each will specify a different `url` for a different combination of `weight`, `style`, etc. parameters.

But the value of `weight` would never be added to a file list...?

Note the comments also don't make sense, it's not "initial" vs. "newer" support, it's just whether the user is providing one vs. multiple faces per family.

Reply:
You can see the equivalent JS font path parsing code here: https://github.com/tangrams/tangram/blob/master/src/scene_loader.js#L128-L138

Reply:
This did look suspicious to me, but I left it according to how it was done before including the comment: https://github.com/tangrams/bundler/blob/4bd7821e4c29aef41bc5a91afe5ac0ba1e0b3cc6/tangram_bundler/__init__.py#L40-L43

Reply:
I should confirm but I also do not remember ES parser supporting urls for `weight`

Reply:
On further investigation, it looks like this code was handling *multiple* faces per font definition (value is an array of face objects), but not *single* face definitions, such as in https://github.com/tangrams/tangram/blob/master/demos/scene.yaml#L38-L39:

```
fonts:
    Montserrat:
        url: fonts/montserrat.woff
```

Pushed a fix in 53e8072a01fb1111a65b611b9b6a57610e9a9b91